src/assets/scripts/client/navigationLibrary/Route/RouteModel.js

```diff
@@ -135,15 +135,28 @@ export default class RouteModel extends BaseModel {
     }
 
     /**
-     * Verify that a routeCode has exactly 3 segments
+     * Verify that a routeString has exactly 3 segments
      *
      * @for RouteModel
      * @method _isValidRouteCode
-     * @param routeCode {string}
+     * @param routeString {string}
      * @return {boolean}
      * @private
      */
-    _isValidRouteCode(routeCode) {
-        return routeCode.split(SEGMENT_SEPARATION_SYMBOL).length === MAXIMUM_ROUTE_SEGMENT_LENGTH;
+    _isValidRouteCode(routeString) {
+        return RouteModel.isProcedureRouteString(routeString);
     }
```

Comment:
Way I see it, the RouteModel should be all encompassing of an entire "route", and the individual pieces are another class. Here, `RouteModel` is really being used like what I would call a `RouteSegmentModel` or `SimpleRouteSegmentModel`, which differs from a `ProcedureRouteSegmentModel`. An array of these things should make up a `RouteModel`.

Maybe I'm just stuck in the terms I am familiar with, but using the word "route" alone and having it not mean the *entire* flight plan route kinda makes my head explode.